src/actions/general.test.js

```diff
@@ -26,30 +25,6 @@ describe('Actions.General', () => {
         await TestHelper.tearDown();
     });
 
-    it('getPing - Invalid URL', async () => {
-        const serverUrl = Client4.getUrl();
-        Client4.setUrl('notarealurl');
-        await Actions.getPing(true)(store.dispatch, store.getState);
```

Comment:
Something like https://github.com/mattermost/mattermost-redux/pull/748/files#diff-6efd7618205d1ef44f19804bf6b2750fR125

Reply:
Take a look at this commit https://github.com/mattermost/mattermost-redux/pull/940/commits/452fff23db7bc859a2579940361fc452a914fd81 is this what you had in mind? For the `getPing - Invalid URL` case I was expecting [`pingError`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L37) but...I get  `undefined` instead.

Reply:
Can you try using a nock like this to return an error and try?

```
nock(Client4.getBaseRoute()).
            get('/system/ping').
            query(true).
            reply(401, {status_code: 401,  error: 'ping error');
```

Reply:
@hmhealey Waiting for one minor change change here. Check my last comment in this thread

Reply:
Hey @sudheerDev here's the situation:

I rewrote this test. It's in [another branch](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general-general.test-debug/src/actions/general.test.js#L28-L47) because [`getPing()`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L19-L42) is not returning what I expect.

If I do something like this...works great!
```javascript
        nock(Client4.getBaseRoute()).
            get('/system/ping').
            query(true).
            reply(200, {status: 'OK', version: '4.0.0'});
```
Try anything else and [`getPing()`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L19-L40) returns `undefined`. 
```javascript
        nock(Client4.getBaseRoute()).
            get('/system/ping').
            query(true).
            reply(401, {error: 'ping error', status_code: 401});
```
But here's the weird part! [`getPing()`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L19-L40) *only* returns the [`{data}`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L40) sent or `undefined`. Nothing else! To test this - I modifed what [`getPing()`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L40) returns...

...hard-coding to always return the error message I want, `undefined`.
```javascript
// return {data}
return {error: pingError}
```
...trying anything else, `undefined`. Returning anything *not* `{data}` is `undefined`.
```javascript
// return {data}
return {'anything':'else'}
```
I'm worried [`getPing()`](https://github.com/JosephSamela/mattermost-redux/blob/12523-remove-unused-status-flags-general/src/actions/general.js#L19-L42) is messed up somehow. Am I missing something obvious?

Reply:
Hey, 
i did this and the test work for me.

`import {FormattedError} from './helpers.js';` at the start
I added this as a test case and it is working for me

```
        const pingError = new FormattedError(
            'mobile.server_ping_failed',
            'Cannot connect to the server. Please check your server URL and internet connection.'
        );

        nock(Client4.getBaseRoute()).
            get('/system/ping').
            query(true).
            reply({error: 'ping error', status_code: 401, code: 401});
        const {error} = await Actions.getPing()(store.dispatch, store.getState);
        assert.deepEqual(error, pingError);
```

let me know if you want me to add changes to the PR.

Reply:
@sudheerDev Thank you for the guidance! I made these changes https://github.com/mattermost/mattermost-redux/pull/940/commits/07de2f065a8cdbfd8c61b2bd455a32466e53471e. I ran `make check-style` `make test` `make flow` and everything passes!